lib/ts/recipe/totp/prebuiltui.tsx

```diff
@@ -0,0 +1,102 @@
+    constructor(public readonly recipeInstance: TOTPRecipe) {
+        super();
+    }
+
+    // Static methods
+    static getInstanceOrInitAndGetInstance(): TOTPPreBuiltUI {
+        if (TOTPPreBuiltUI.instance === undefined) {
+            const recipeInstance = TOTPRecipe.getInstanceOrThrow();
+            TOTPPreBuiltUI.instance = new TOTPPreBuiltUI(recipeInstance);
+        }
+
+        return TOTPPreBuiltUI.instance;
+    }
+    static getFeatures(
+        useComponentOverrides: () => GenericComponentOverrideMap<any> = useRecipeComponentOverrideContext
+    ): RecipeFeatureComponentMap {
+        return TOTPPreBuiltUI.getInstanceOrInitAndGetInstance().getFeatures(useComponentOverrides);
+    }
+    static getFeatureComponent(
+        componentName: "mfaTOTP",
+        props: any,
+        useComponentOverrides: () => GenericComponentOverrideMap<any> = useRecipeComponentOverrideContext
+    ): JSX.Element {
+        return TOTPPreBuiltUI.getInstanceOrInitAndGetInstance().getFeatureComponent(
+            componentName,
+            props,
+            useComponentOverrides
+        );
+    }
+
+    // Instance methods
+    getFeatures = (
+        useComponentOverrides: () => GenericComponentOverrideMap<any> = useRecipeComponentOverrideContext
+    ): RecipeFeatureComponentMap => {
+        const features: RecipeFeatureComponentMap = {};
+        if (this.recipeInstance.config.totpMFAScreen.disableDefaultUI !== true) {
+            const normalisedFullPath = this.recipeInstance.config.appInfo.websiteBasePath.appendPath(
+                new NormalisedURLPath(DEFAULT_TOTP_PATH)
+            );
+            features[normalisedFullPath.getAsStringDangerous()] = {
+                matches: matchRecipeIdUsingQueryParams(this.recipeInstance.config.recipeId),
+                component: (props: any) => this.getFeatureComponent("mfaTOTP", props, useComponentOverrides),
+                recipeID: TOTPRecipe.RECIPE_ID,
+            };
+        }
+        return features;
+    };
+    getFeatureComponent = (
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        _: "mfaTOTP",
+        props: any,
+        useComponentOverrides: () => GenericComponentOverrideMap<any> = useRecipeComponentOverrideContext
+    ): JSX.Element => {
+        return (
+            <UserContextWrapper userContext={props.userContext}>
+                <SessionAuth overrideGlobalClaimValidators={() => []}>
+                    <MFATOTPFeature
+                        recipe={this.recipeInstance}
+                        useComponentOverrides={useComponentOverrides}
+                        {...props}
+                    />
+                </SessionAuth>
+            </UserContextWrapper>
+        );
+    };
+
+    // For tests
+    static reset(): void {
+        if (!isTest()) {
+            return;
+        }
+
+        TOTPPreBuiltUI.instance = undefined;
+        return;
+    }
+
+    static MFATOTP = (props?: any): JSX.Element =>
```

Comment:
we need to correctly type this, and also make sure that it has the navigation prop (just like other recipes in the master branch)